src/main/java/com/minecolonies/coremod/items/ItemResourceScroll.java

```diff
@@ -70,14 +71,24 @@ public ActionResultType useOn(ItemUseContext ctx)
 
         if (entity instanceof TileEntityColonyBuilding)
         {
-            compound.putInt(TAG_COLONY_ID, ((AbstractTileEntityColonyBuilding) entity).getColonyId());
-            BlockPosUtil.write(compound, TAG_BUILDER, ((AbstractTileEntityColonyBuilding) entity).getPosition());
-
-            if (!ctx.getLevel().isClientSide)
-            {
-                LanguageHandler.sendPlayerMessage(ctx.getPlayer(),
-                  TranslationConstants.COM_MINECOLONIES_SCROLL_BUILDER_SET,
-                  ((AbstractTileEntityColonyBuilding) entity).getColony().getName());
+            final AbstractTileEntityColonyBuilding buildingEntity = (AbstractTileEntityColonyBuilding) entity;
+
+            if (buildingEntity.getBuilding() instanceof BuildingBuilder) {
```

Comment:
Please adjust the formatting to fit our style. You can find a formatting file in the documentation folder in the REPO.
Manual formatting would be then

{


}